ui/src/pages/network/Fees.tsx

```diff
@@ -0,0 +1,140 @@
+
+  useEffect(() => {
+    if (!feeSchedule) return;
+    setAssetLabel(feeSchedule.payload.currentFee.currency.label);
+    setQuantity(feeSchedule.payload.currentFee.amount);
+    setAccountLabel(feeSchedule.payload.feeAccount.id.label);
+  }, [feeSchedule]);
+
+  const assets = useStreamQueries(AssetDescription).contracts;
+
+  const asset = assets.find(c => c.payload.assetId.label === assetLabel);
+  const assetSettlementRules = useStreamQueries(AssetSettlementRule).contracts;
+  const accounts = assetSettlementRules
+    .map(c => c.payload.account)
+    .filter(acc => acc.owner === party);
+  const account = accounts.find(a => a.id.label === accountLabel);
+  const accountRule = assetSettlementRules.find(ar => ar.payload.account.id.label === accountLabel);
+
+  const canRequest = !feeSchedule
+    ? !!assetLabel && !!asset && !!accountLabel && !!account && !!quantity
+    : !!quantity;
+
+  if (!role)
+    return (
+      <div>
+        <h2>Party "{party}" can not setup a Fee Schedule.</h2>
+      </div>
+    );
+
+  const createFeeSchedule = async () => {
+    if (!feeSchedule) {
+      if (!asset || !account) return;
+      await ledger.exercise(ExchangeRole.CreateFeeSchedule, role.contractId, {
+        currency: asset.payload.assetId,
+        feeAccount: account,
+        quantity,
+      });
+    } else {
+      if (!asset) return;
+      await ledger.exercise(FeeSchedule.UpdateFeeSchedule, feeSchedule.contractId, {
+        amount: quantity,
+        currency: asset.payload.assetId,
+      });
+    }
+  };
+
+  return (
+    <div className="page-section-row">
+      <Tile className="inline" header="Setup Fees">
+        <br />
+        {!schedulesLoading && (
+          <FormErrorHandled onSubmit={createFeeSchedule}>
+            {!feeSchedule && (
+              <>
+                <Form.Select
+                  selection
+                  label="Fee Account"
+                  options={accounts.map(c => ({
+                    text: c.id.label,
+                    value: c.id.label,
+                  }))}
+                  value={accountLabel}
+                  onChange={(_, d) => setAccountLabel((d.value && (d.value as string)) || '')}
+                />
+              </>
+            )}
+            <Form.Select
+              selection
+              label="Currency"
+              options={assets.map(c => ({
+                text: c.payload.assetId.label,
+                value: c.payload.assetId.label,
+              }))}
+              value={!!feeSchedule ? assetLabel : assetLabel}
+              onChange={(_, d) => setAssetLabel((d.value && (d.value as string)) || '')}
+            />
+
+            <Form.Input
+              placeholder={!!feeSchedule ? feeSchedule.payload.currentFee.amount : quantity}
+              required
+              fluid
+              number
+              label="Fee Amount"
+              value={quantity}
+              onChange={e => setQuantity(e.currentTarget.value)}
+            />
```

Comment:
Is there a common percentage/suggested value or something we can use here? I am just thinking about what eric said about raw input fields. Would be cool if you could specify a percentage or an amount. Just a thought.

Reply:
I'm not sure what you mean by percentage, this is for a flat fee (although a percentage fee would be interesting)...maybe I will just default it to `0.0` so it at least there is something there.

Reply:
0.0 sounds good - I was thinking the fee was a percentage of some traded quantity. Or it could be optionally represented that way. But since it is a flat fee I guess that wouldnt make much sense!